lib/UdpConnection.js

```diff
@@ -14,13 +14,17 @@ module.exports = function factory (userOptions) {
   const socket = dgram.createSocket('udp4')
 
   const writableStream = new stream.Writable({
-    close () { socket.close() },
     write (data, encoding, callback) {
       socket.send(data, 0, data.length, options.port, options.address)
       callback()
     }
   })
 
-  process.stdin.pipe(writableStream, {end: false})
+  writableStream.close = function () {
+    socket.unref()
```

Comment:
why this `unref()` is being added?

Reply:
The same reason as in UDP connection.

Reply:
`unref()` does not do what you think it does. It just enables the process to exit if if the socket is open.
It should be removed

Reply:
But how in another way exit the process and successfully send all of the messages?

Reply:
we need to close the socket.

Reply:
But as I said before in my case it did not send all of the messages successfully. Some of them were not sent.

Reply:
That's down to the way the OS handles processes and sockets. Your application needs to implement a graceful shutdown if it is critical that all messages being piped out are captured.

Reply:
Got it. ```unref``` is not a good solution for this and it will be removed. Maybe you can recommend some other way how to graceful shutdown the socket?

Reply:
That's very specific to your application. I don't know how I would be able to even guess at it.